### ISSUE TITLE
aCompCor components with different masks are inter-mixed

### ISSUE THREAD

**Issue by effigies** [nipreps/fmriprep#2384]
Going by https://neurostars.org/t/fmriprep-acompcor-5-csf-and-5-wm/18526, it seems we're not segregating different aCompCor masks in the regressors file? At one point there had been talk of `[acwt]_comp_cor_XX` to distinguish aCompCor (WM + CSF), cCompCor (CSF), wCompCor (WM) and tCompCor (temporal). Is there a reason we didn't do that?

**Comment by oesteban** [comment]
@rciric do you have any quick ideas on what could be going on here?

**Comment by pbellec** [comment]
Quick heads up: in [load_confonuds](https://github.com/SIMEXP/load_confounds) we are now parsing the json to properly handle anatCompCor without mixing the masks. But this is very counter-intuitive and hard to follow for new users. The previous version of load_confounds did mix components from the combined, WM and CSF masks...

I would vote for integrating the mask in the tsv confound labels in the future.

**Comment by effigies** [comment]
Putting this on the queue for next release, if we can manage it, since it's a derivatives change.

I propose reverting to `a_comp_cor` for CSF+WM, `w_comp_cor` for WM and `c_comp_cor` for CSF.

**Comment by njr175** [comment]
> Putting this on the queue for next release, if we can manage it, since it's a derivatives change.
> 
> I propose reverting to `a_comp_cor` for CSF+WM, `w_comp_cor` for WM and `c_comp_cor` for CSF.

Really hope this change occurs in the next release; This would be a very helpful change. In the meantime, does anyone have a suggestion for the best way to work with the fMRIPrep .tsv file which houses these compcor regressors? We want to pull and use the first 5 aCompCors from CSF, and the first 5 from WM, but with the data set that I am currently working with, they are pretty different in their numbering in the corresponding .json file (e.g., aCompCorr_000 sometimes pulls from CSF, but for other participants it pulls from Combined). Thanks for any suggestions.

**Comment by pbellec** [comment]
@njr175 I believe https://github.com/simexp/load_confounds deals with it appropriately, by looking at the json information and reordering the components. If you think it does not behave correctly it would be very useful to know. Note however that the project is about to get merged in nilearn with some pretty significant changes in the API.

**Comment by njr175** [comment]
@pbellec Thanks so much! I will look into this. If I have any issues I will let you know! Thanks again.

**Comment by effigies** [comment]
I hadn't planned to, since it breaks derivatives. It could be done with an off-by-default flag.

**Comment by HippocampusGirl** [comment]
Thanks for the quick response! I guess it makes more sense to adapt the code that reads in the confounds then, especially as this has been made quite easy with the `load_confounds` function linked above.
